[feat] accept PUBLIC as prefix for environment variables (#2355)
sveltejs/kit: documentation/faq/60-env-vars.md

question: How do I use environment variables?
---

> Vite uses [dotenv](https://github.com/motdotla/dotenv) to load environment variables from a file named `.env` or similar. By default, Vite only exposes environment variables prefixed with `VITE_`. However, SvelteKit overrides Vite's default `envPrefix` setting to also allow `PUBLIC_` as a prefix. You would need to instantiate dotenv yourself if you want all environment variables exposed in `process.env['YOUR_ENV_VAR']`. We hope to see all environment variables exposed on the server-side [in the future](https://github.com/vitejs/vite/issues/3176).

Reviewer: @dummdidumm I don't think this is breaking, but maybe it can be phrased more friendlier here

```suggestion
Vite uses [dotenv](https://github.com/motdotla/dotenv) to load environment variables from a file named `.env` or similar. SvelteKit allows you to use any environment variables prefixed with either `VITE_` or `PUBLIC_`. You would need to instantiate dotenv yourself if you want all environment variables exposed in `process.env['YOUR_ENV_VAR']`. We hope to see all environment variables exposed on the server-side [in the future](https://github.com/vitejs/vite/issues/3176).
```

Reviewer 2: Ah I misread the code - the allowed prefixes are now both vite and client, right?

Reviewer: Yeap

Reviewer: Wait no, it's vite and public

Author: I updated the text similar to your suggestion, but also mentioning `envPrefix` still

Correct that this allows both to avoid being breaking
